### ISSUE TITLE
Proposal: Use standard button order in ContentDialog

### ISSUE THREAD

**Issue by Jaiganeshkumaran** [microsoft/microsoft-ui-xaml#4034]
# Proposal: Use standard button order in ContentDialog
For a long time, Windows has used a system where the primary button is kept first which is not the standard on most other platforms.

## Summary

To be consistent with the web and other platforms, consider changing the button order in ContentDialog from Primary, Secondary, and Close to Secondary, Primary, and close.

## Rationale

* Users expect a familiar system across devices 
## Scope

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow developers to have a familiar dialog system out of the box on Windows | Must |
| Developers using ContentDialog will see this change automatically if they used Primary, Secondary and close buttons and will not require any source code change unless they are using custom control | Should |

**Comment by huoyaoyuan** [comment]
Primary button first is **standard on Windows**, with a history much longer than mobile systems and maybe some desktop environment.

WinUI is **native Windows component** and should follow Windows everywhere.

**Comment by stmoy** [comment]
Hi @Jaiganeshkumaran - thanks for filing this issue. You're correct to observe that the order of the buttons in dialogs are different on other platforms compared to Windows. This is a multi-faceted issue, but it boils down to what @huoyaoyuan mentions: that the order of the dialog buttons in Windows should be consistent across Windows. 

Part of the reason is history (ie. long-time Windows users are familiar with this order since it has been like this since the early days of Windows); part of this is accessibility (ie. low-vision users have an expectation for the dialog button order when using screen readers on Windows); part of this is technology (ie. disparate technologies across Windows including Xaml, DUI, WPF, WinForms, Html, etc. use the standard Windows order, so fixing this in WinUI alone would not be enough to move the entire ecosystem of Windows apps to have the new order). 

In summary: the discrepancy is known, but deliberate. Closing this since we're not going to change this behavior.

**Comment by Jaiganeshkumaran** [comment]
@stmoy What's this then? Please tell everyone to follow the guidelines
![திரைப்பிடிப்பு (232)](https://user-images.githubusercontent.com/45227503/106243023-a15a4f00-622e-11eb-8632-9a87c3be3ca1.png)

**Comment by stmoy** [comment]
@Jaiganeshkumaran - thanks for bubbling this up. Even though our intent is to unify the button order across all Windows experiences, not every team does the right thing - and sometimes, there are discrepancies within the same experience.

For example, the website has the right order:

![image](https://user-images.githubusercontent.com/25991996/106317967-ec229980-6223-11eb-92f8-6782111e8586.png)

But the app has the wrong order:

![image](https://user-images.githubusercontent.com/25991996/106318803-4d973800-6225-11eb-8ce6-643055fa31eb.png)

😱

In this case, the Whiteboard app doesn't appear to be a WinUI app in the first place (it looks to me like a PWA, but I'm just speculating). Part of the reason that we don't want to change the order in WinUI is because it would increase the fragmentation across - not decrease it, or at least, not decrease it for a very long time as it would take significant time for app teams to update.
